### PR TITLE
add rangefinder

### PR DESCRIPTION
**Pull request by AsiiaPine**
The module has its own separated parameters: scale, offset and enable (scale and offset was used since the measurements taken in the lab had some shift)
Timing:
- parameters update is performed with 1 Hz freq.
- data collection — 10 Hz
- data publication — 10Hz

image

### CODE REVIEW

File: Src/dronecan_application/modules/RangeFinder/params.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  max: 40000
+
+range_finder.scale:
+  type: Integer
+  note: Offset for range finder in ptc [%].
+  enum: PARAM_RANGE_FINDER_SCALE
+  flags: mutable
+  default: 100
+  min: -1000
+  max: 1000
```
Comment (PonomarevDA) [inline Src/dronecan_application/modules/RangeFinder/params.yaml:17]: Let's keep only 2 parameters here at the moment:
- range.enable
- range.id

Offset parameter can be useful, but it is actually slightly more complicated that a single parameter. An autopilot needs pos_x, pos_y, pos_z, pitch, roll values to calculate the flight height. This configuration is done on the PX4 side with EKF2_RNG_* parameters. More details: https://docs.px4.io/main/en/sensor/rangefinders.html#configuration

No doubt, we can duplicate these parameters here and apply some math to calculate the required distance, but it requires a few times more lines of code and I'm not sure we will even use it some day. PX4 parameters are sufficient for us.

As I see the scale parameter is redundant here. Usually, a sensor estimate the distance quite precisely withing his operating conditions and doesn't need an additional calibration.

File: Src/dronecan_application/modules/RangeFinder/params.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+range_finder.enable:
+  type: Integer
+  note: Indicates the operational mode of the node. When set to 1, the uavcan.equipment.range_sensor will be transmitted. 
+  enum: PARAM_RANGE_FINDER_ENABLE
+  flags: mutable
+  default: 0
```
Comment (PonomarevDA) [inline Src/dronecan_application/modules/RangeFinder/params.yaml:24]: Let's enable it by default

**Review by PonomarevDA — CHANGES_REQUESTED** [review]
The timestamp has a random value. It is serialized incorrectly. Everything else seems fine.

```c++
uint8_t offset = 0;
auto timestamp = obj->timestamp;
canardEncodeScalar(buffer, 56, offset, &timestamp);
offset += 56;
```

File: Src/dronecan_application/modules/RangeFinder/params.yaml (hunk 1 above)
Comment (AsiiaPine) [reply Src/dronecan_application/modules/RangeFinder/params.yaml]: Done

**Review by PonomarevDA — CHANGES_REQUESTED** [review]
Good. Please, just update the submodule to the main branch because `This commit does not belong to any branch on this repository`.

**Review by PonomarevDA — APPROVED** [review]

**Pull request merged**